File: llama2/php/119.php
<?php
function create_file($path, $filename, $content, $overwrite=false) {
  $res = false;
  if ($path == '/' || strpos($path, '..') !== false) return false;
  // Check for invalid file name
  if (preg_match('/^[a-z0-9\\.]+$/i', $filename)) {
    // Check for existing file
    if ($overwrite === true || !file_exists($path . '/' . $filename)) {
      // Open file
      if ($f = @fopen($path . '/' . $filename, 'w')) {
        fwrite($f, $content);
        fclose($f);
        $res = true;
      }
    }
  }
  return $res;
}

// Test the function
$path = './';
$filename = 'test.txt';
$content = 'Hello, World!';
$overwrite = true;

echo create_file($path, $filename, $content, $overwrite) ? 'File created successfully!' : 'Failed to create file!';
?>

File: llama2/python/119.py
import base64
import os
from cryptography.fernet import Fernet

def main():
    # Read the key from the file
    with open('key.txt', 'rb') as f:
        encoded_key = f.read()

    # Decode the key
    key = base64.b64decode(encoded_key)

    # Check the length of the decoded key
    if len(key) != 32:
        raise ValueError('Invalid Fernet key length')

    # Create the encryption object
    ecy = Fernet(key)

if __name__ == "__main__":
    main()